client/sources/common/interpreter.py

```diff
@@ -339,7 +340,7 @@ def _compare(self, expected, code):
             print('# but got')
             print('\n'.join('#     {}'.format(line)
                             for line in actual.output_lines()))
-            if not self.show_all_cases:
+            if not self.show_all_cases or actual.exception_type == exceptions.Timeout.__name__:
```

Comment:
I think a comment here might help future code reader, like
"# Bail out on first failed test, or if we're showing all test results, bail on loop timeout
"

Reply:
Does this need a check for actual_exception? (I noticed the earlier condition is (actual.exception  and [actual.exception_type] == expected.output_lines()) 
and just want to double check we don't need to check actual.exception first)

Reply:
It seems to me that we don't need it. I'm not sure why it's included in the above code, because exception and exception_type should both be initialized to None https://github.com/tom11my/ok-client/blob/fpp-timeout-fix/client/sources/common/interpreter.py#L375-L376

Reply:
hmm, for consistency with existing code, i would suggest checking `actual.exception` here too -- consistency makes things easier for future readers

Reply:
sounds good.